plugins/plugins.go

```diff
@@ -160,10 +160,11 @@ type StatusListener func(status map[string]*Status)
 // Manager implements lifecycle management of plugins and gives plugins access
 // to engine-wide components like storage.
 type Manager struct {
-	Store  storage.Store
-	Config *config.Config
-	Info   *ast.Term
-	ID     string
+	Store           storage.Store
+	Config          *config.Config
+	BootstrapConfig *config.Config
```

Comment:
Nit: Do we need to export this field? Also can we just store the bootstrap labels ie `bootstrapConfigLabels` and in the manager initialization we would have:

```go
m := &Manager{
		Store:                        store,
		Config:                       parsedConfig,
		ID:                           id,
		keys:                         keys,
		pluginStatus:                 map[string]*Status{},
		pluginStatusListeners:        map[string]StatusListener{},
		maxErrors:                    -1,
		interQueryBuiltinCacheConfig: interQueryBuiltinCacheConfig,
		serverInitialized:            make(chan struct{}),
                bootstrapConfigLabels:        parsedConfig.Labels,
	}
```

Reply:
Sure, will make this private. About making it more specific: when looking at overriding discovery with local changes (another issue), I felt that keeping the bootstrap config was really useful so I think I got my wires crossed. Will make that change for now, can be extended later...